Source/Authorization/OAuthService.swift

```diff
@@ -87,7 +83,23 @@ class OAuthService {
         }
     }
     
-    private func extractPeAuthorizationCode(from response: PreAuthResponse, completion: @escaping (Result<String, Error>) -> Void) {
+    func requestReferenceToken(oauthToken: OAuthToken, completion: @escaping (Result<TokenSessionResponse, Error>) -> Void) {
+        guard let jwt = JWTUtility.dataTriggerRequestJWT(accessToken: oauthToken.accessToken.value, configuration: configuration) else {
+            fatalError("Invalid reference token request JWT")
```

Comment:
We should probably steer away from fatal errors in the context of the SDK, because this will cause the host application to exit.

Reply:
Yeah - we haven't defined errors yet or have a strategy as to what errors to present to dev - I thought this would only fail due to incorrect dev setup and wouldn't happen during production run (once dev has set everything up correctly). However, it probably is safest to define an SDK error for this and pass up the chain to caller.